### PR TITLE
Add the well-used htmlwriter.py script to a repo.

### PR DESCRIPTION
**Pull request by doctoryes**
https://openedx.atlassian.net/browse/PLAT-1752
Also, add support for the new pytest format while keeping the support for the nose format functional.

### CODE REVIEW

**Comment by pwnage101** [comment]
Remember to update the wiki after merge: https://github.com/edx/edx-tools/wiki

File: test-reports/htmlwriter.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+def clipped(text, maxlength=150):
+    """Return the text, but at most `maxlength` characters."""
+    if len(text) > maxlength:
+        text = text[:maxlength-1] + u"\N{HORIZONTAL ELLIPSIS}"
+    return text
+
+
+def report_file(path, html_writer):
+    """Report on one nosetests.xml file."""
```
Comment (pwnage101) [inline test-reports/htmlwriter.py]: how about "report on one test result xml file".

File: test-reports/htmlwriter.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+    html_writer.end_section()
+    return results
+
+
+def main(start):
+    totals = TestResults()
+    html_writer = HtmlOutlineWriter(sys.stdout)
+    for dirpath, _, filenames in os.walk(start):
+        if "nosetests.xml" in filenames:
```
Comment (pwnage101) [inline test-reports/htmlwriter.py]: I think a note explaining that pytest currently writes files called "nosetests.xml" would be helpful.

File: test-reports/htmlwriter.py (hunk 1 above)
Comment (doctoryes) [reply test-reports/htmlwriter.py]: Done.

File: test-reports/htmlwriter.py (hunk 2 above)
Comment (doctoryes) [reply test-reports/htmlwriter.py]: Done.

**Comment by nedbat** [comment]
I know the filename is my fault, but it's a really bad name. htmlwriter? that hardly summarizes what this thing does.  summarize_tests.py? rollup_tests.py? Something like that?

**Comment by doctoryes** [comment]
Glad to hear you aren't attached to the name, @nedbat ! I also vote for a name change - and will do so.

**Comment by doctoryes** [comment]
I went with `summarize_test_results.py` - wordy but accurate.

**Comment by doctoryes** [comment]
@pwnage101 Added script to wiki.

**Pull request merged**